### ISSUE TITLE
[a11y]: API key modal, Screen reader identifying hidden controls

### ISSUE THREAD

**Issue by anamikaanu96** [carbon-design-system/ibm-products#6247]
### Package

@carbon/ibm-products

### Browser

Firefox

### Operating System

Windows

### Package version

v2.52.0-rc.0

### Automated testing tool and ruleset

Manual testing

### Assistive technology

JAWS

### Description

Screen reader  identifying - "a focus sentinel button" on both Top of Modal which is visually not visible :
1. On Arrow key navigation  this Element identified  both on Top part of the Modal.
2.  "Focus sentinel" Button is not keyboard  operable.
3. Existing Label of this control is not descriptive  enough to understand  the purpose of this Element.

Expected behaviour : Please confirm,  the functional purpose of this "focus sentinel button"
1. If there is any  functional objective , Make this Element visually available , also operable by keyboard /mouse as well as give a  descriptive Label which explain its purpose .
2. If not, Remove focus also hide it from screen reader detection on both Tab/Arrow key navigation.

### Reproduction/example

https://ibm-products.carbondesignsystem.com/?path=/story/ibm-products-components-generating-an-api-key-apikeymodal--generate&globals=viewport:basic

### Steps to reproduce

This was reported in all variants of API Key Modal.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems
